include/dba.php

```diff
@@ -355,73 +429,98 @@ public function dbg($dbg) {
 
 	public function escape($str) {
 		if ($this->db && $this->connected) {
-			if ($this->mysqli) {
-				return @$this->db->real_escape_string($str);
-			} else {
-				return @mysql_real_escape_string($str,$this->db);
+			switch ($this->driver) {
+				case 'pdo':
+					return substr(@$this->db->quote($str, PDO::PARAM_STR), 1, -1);
```

Comment:
I intentionally haven't changed too much. Only the PDO part is new, the code for the other drivers simply was moved. This was done to not cause any unwanted side effects.